### PR TITLE
Ignore PRs for categories ME and PM

### LINKED ISSUE #25: [Core Weekly] Ignore PRs labeled "Merge" and "Project Management" (ME and PM) in core weekly

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Ignore PRs for categories ME and PM
| Type?         | bug fix
| Fixed ticket? | Fixes https://github.com/PrestaShop/core-weekly-generator/issues/25

### CODE REVIEW

File: core_weekly/core_weekly.py [hunk 1]
```diff
@@ -94,6 +94,9 @@ def generate(self):
         :rtype: str
 
         """
+
+        category_blacklist = ['ME', 'PM']
```
Comment (PierreRambaud) [inline core_weekly/core_weekly.py]: Must be in `coreteam`:
`from coreteam import CATEGORY_BLACKLIST`

File: core_weekly/template.py [hunk 2]
```diff
@@ -288,7 +288,7 @@ def custom_sort(self, data, key_order):
 
         return sorted_dict
 
-    def build_merged_pull_requests(self, result):
+    def build_merged_pull_requests(self, result, category_blacklist):
```
Comment (PierreRambaud) [inline core_weekly/template.py]: No need to be in the parameters. use
`from coreteam import CATEGORY_BLACKLIST`

File: core_weekly/template.py [hunk 3]
```diff
@@ -315,6 +315,9 @@ def build_merged_pull_requests(self, result):
             for category, items in sorted_category_items.items():
                 category_name = CATEGORIES[category] if category in CATEGORIES.keys() else category
 
+                if (category in category_blacklist):
```
Comment (PierreRambaud) [inline core_weekly/template.py]:
```suggestion
                if category in CATEGORY_BLACKLIST:
```
There is no parenthesis in python (if not needed)

**Comment by matks** [comment]
@PierreRambaud Feedback adressed

File: core_weekly/core_weekly.py [hunk 4]
```diff
@@ -128,7 +126,7 @@ def generate(self):
             merged_pull_requests,
             self.date_range
         )
-        content += self.template.build_merged_pull_requests(merged_pull_requests, category_blacklist)
+        content += self.template.build_merged_pull_requests(merged_pull_requests, CATEGORIES_REJECT_LIST)
```
Comment (PierreRambaud) [inline core_weekly/core_weekly.py]:
```suggestion
        content += self.template.build_merged_pull_requests(merged_pull_requests)
```

File: core_weekly/core_weekly.py [hunk 5]
```diff
@@ -4,6 +4,7 @@
 from .template import Template
 from .parser import Parser
 from pathlib import Path
+from coreteam import CATEGORIES_REJECT_LIST
```
Comment (PierreRambaud) [inline core_weekly/core_weekly.py]:
```suggestion
```

File: core_weekly/template.py [hunk 6]
```diff
@@ -315,7 +315,7 @@ def build_merged_pull_requests(self, result, category_blacklist):
             for category, items in sorted_category_items.items():
                 category_name = CATEGORIES[category] if category in CATEGORIES.keys() else category
 
-                if (category in category_blacklist):
+                if category in category_blacklist:
```
Comment (PierreRambaud) [inline core_weekly/template.py]:
```suggestion
                if category in CATEGORIES_REJECT_LIST:
```

Comment (PierreRambaud) [reply core_weekly/template.py]: It's a const, it doesn't need to be injected in the template. 
Do like CATEGORIES, it's enough

Comment (matks) [reply core_weekly/template.py]: Would'nt it be better to keep `build_merged_pull_requests()` configurable (so we can pass whatever denylist we want to it) ?

If for example I want to unit-test `build_merged_pull_requests()` ; by using a parameter I can check different scenarii

Comment (PierreRambaud) [reply core_weekly/template.py]: This is something you're already able to do with Python, it helps you to patch constant:
https://docs.python.org/3/library/unittest.mock.html#where-to-patch

Comment (matks) [reply core_weekly/template.py]: Thank you for the insights 😄 always great to learn

I will apply the feedback

**Comment by PierreRambaud** [comment]
Thank you @matks

**Pull request merged**